Add plugin: TOC compatible with Publish

BrMacCath opened the pull request:
# Repo URL
Link to my plugin: https://github.com/BrMacCath/Table-of-Contents

# Release Checklist
- [x] I have tested the plugin on.
    - [x] Windows
    - [ ] macOS
    - [ ] Linux
    - [ ] Android (if applicable)
    - [ ] iOS (if applicable)
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
    - [x]  main.js
    - [x]  manifest.json
    - [ ]  styles.css (optional)
- [x] GitHub release name matches the exact version number specified in my manifest.json (Note: Use the exact version number, don't include a prefix v)

- [x] The id in my manifest.json matches the id in the community-plugins.json file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x]I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
I have given proper attribution to these other projects in my README.md.

joethei commented:
Some screenshots in the README would be nice.

["description": "This generates a table of contents that is compatible with publish.",](https://github.com/BrMacCath/Table-of-Contents/blob/88d6c65b5d357121a30405c6a4dfe746fea49ac9/manifest.json#L6)
Don't start your description with "This does xyz", it's getting a bit repetetive, as a lot of plugins already use something like that.

[const fileContent = await this.app.vault.read(file);](https://github.com/BrMacCath/Table-of-Contents/blob/88d6c65b5d357121a30405c6a4dfe746fea49ac9/main.ts#L111-L122)
You should use the [`Vault.process`](https://docs.obsidian.md/Reference/TypeScript+API/Vault/process) API for the file modification

[name: "Create Table Of Contents",](https://github.com/BrMacCath/Table-of-Contents/blob/88d6c65b5d357121a30405c6a4dfe746fea49ac9/main.ts#L144)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

[this.app.workspace.on("active-leaf-change", async () => {](https://github.com/BrMacCath/Table-of-Contents/blob/88d6c65b5d357121a30405c6a4dfe746fea49ac9/main.ts#L150)
Events need to be registered, so they can be properly unloaded when the plugin is disabled. <https://docs.obsidian.md/Plugins/Events>

[console.log("Reran TOC");](https://github.com/BrMacCath/Table-of-Contents/blob/88d6c65b5d357121a30405c6a4dfe746fea49ac9/main.ts#L159)
Please avoid unnecessary logging. If you need it for debugging purposes, add a check to only log during development.

BrMacCath commented:
> Some screenshots in the README would be nice.
> 
> ["description": "This generates a table of contents that is compatible with publish.",](https://github.com/BrMacCath/Table-of-Contents/blob/88d6c65b5d357121a30405c6a4dfe746fea49ac9/manifest.json#L6) Don't start your description with "This does xyz", it's getting a bit repetetive, as a lot of plugins already use something like that.
> 
> [const fileContent = await this.app.vault.read(file);](https://github.com/BrMacCath/Table-of-Contents/blob/88d6c65b5d357121a30405c6a4dfe746fea49ac9/main.ts#L111-L122) You should use the [`Vault.process`](https://docs.obsidian.md/Reference/TypeScript+API/Vault/process) API for the file modification
> 
> [name: "Create Table Of Contents",](https://github.com/BrMacCath/Table-of-Contents/blob/88d6c65b5d357121a30405c6a4dfe746fea49ac9/main.ts#L144) [Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)
> 
> [this.app.workspace.on("active-leaf-change", async () => {](https://github.com/BrMacCath/Table-of-Contents/blob/88d6c65b5d357121a30405c6a4dfe746fea49ac9/main.ts#L150) Events need to be registered, so they can be properly unloaded when the plugin is disabled. https://docs.obsidian.md/Plugins/Events
> 
> [console.log("Reran TOC");](https://github.com/BrMacCath/Table-of-Contents/blob/88d6c65b5d357121a30405c6a4dfe746fea49ac9/main.ts#L159) Please avoid unnecessary logging. If you need it for debugging purposes, add a check to only log during development.

Hi Joethi,

I have implemented the suggested changes. Thank you.

joethei commented:
You have not updated the description.

BrMacCath commented:
> You have not updated the description.

Hi sorry I forgot about this until I got the email today. I updated it just now. I did it by looking at other peoples' descriptions and imitating them. My idea was to split the description into two sentences. First, describe the action. The second talks about compatibilities. I also removed the "this does" format and just stated what it does without self-referencing itself. Thanks for reviewing this!

joethei commented:
[obsidian publish](https://github.com/BrMacCath/Table-of-Contents/blob/e7e31b13e39150da93758e9b29797c75af82771e/manifest.json#L6)
Please use the correct capitalization, which is "Obsidian Publish"

BrMacCath commented:
Hi @joethei, thanks! I have corrected it there.

BrMacCath commented:
> #### Hello!
> **I found the following issues in your plugin submission**
> 
> **Errors:**
> 
> ❌ Your latest Release is missing the `main.js` file. ❌ Your latest Release is missing the `manifest.json` file.
> 
> This check was done automatically. Do NOT open a new PR for re-validation. Instead, to trigger this check again, make a change to your PR and wait a few minutes, or close and re-open it.

Change to pull request done with this?

BrMacCath commented:
Going to close this for revalidation.

BrMacCath commented:
Reopening this for revalidation.

BrMacCath commented:
Hi @joethei ,

Thanks for all the help along the way. I think I am almost done. My last issue is trying to resolve the conflict with the community JSON file. I wanted to resolve it by putting the plugin at the end of the file, but that has not addressed the issue. I am searching online for a way to fix that, but any help would be appreciated.

BrMacCath reviewed (commented):
Trying to add my plugin table of contents.

BrMacCath commented:
Hi @joethei, I have been trying to fix this issue for an hour every so often. I couldn't see what the issue was, but I found this post https://forum.obsidian.md/t/plugin-validation-fails-couldnt-find-remote-ref/78863 where someone else had the same issue I had, and the resolution was that it was added manually. Could you let me know if I am wrong, and I will spend a few hours this weekend fixing it?

joethei commented:
You have to do a rebase with a force push afterwards to fix it.

BrMacCath commented:
Hi @joethei ,

Thanks for all the help! I think it's pulled in now. First time doing something like this.
